casper/src/main/resources/PoS.rhox

```diff
@@ -340,6 +344,52 @@ new  PoS,
                   }
                 }
               }
+            } |
+
+            contract PoS(@"commitRandomImage", @hash, ackCh) = {
+              new userCh, mvarCh in {
+                getUser!(*userCh) |
+
+                runMVar!(*stateCh, *mvarCh, *ackCh) |
+                for (@validator <- userCh;
+                     @state, resultCh <- mvarCh) {
```

Comment:
Don't use a join here: they always cause merge conflicts and it's not needed here.

Reply:
The conflicts will come from reading the state from the stateCh. Does the system process getUser affect the mergeability?

Reply:
Sorry for the delayed response; I was traveling yesterday.  Reading from the state will cause contention with other blocks trying to read from the state, but a join will cause contention with *every* other block.  

I don't know how system processes affect mergeability.  Is the definition of getUser injected somehow?  I only see sends to the name.

Reply:
According to @lukasz-golebiewski this join here should be fine. See: https://github.com/rchain/rchain/pull/2724#discussion_r326368279

Reply:
If joins only cause conflicts on the names being used, then this is fine.  I thought that joins conflict with every block because you can set up weird chains where only certain of them are allowed to happen at the same time.